### ISSUE TITLE
FMU generation does not return name of global goto Tags

### ISSUE THREAD

**Issue by TaceaCalin1** [CATIA-Systems/FMIKit-Simulink#258]
I am trying to build an FMU out of a subsystem and I am not sure if this is not yet implemented or there is a setting in FMI Kit that I am not using, but the global goto tags are replaced in the FMU with outports, but instead of having the goto tag name, I have outport1, outport2 and so on.

**Comment by t-sommer** [comment]
Can you share a model and steps to reproduce the problem?

**Comment by TaceaCalin1** [comment]
In TestSubsystemNoFMU I placed 3 froms, 3 gains and 3 terminators, just as an example. In TestSubsystem I have 3 constants to 3 gotos. This 3 gotos are set to global and have as corresponding blocks the 3 forms from the TestSubsystemNoFMU. The problem is that when I create the FMU (as you can see in the 3rd block – FMU), the names of the outports don’t have the name of the gotos. They are named by default with outport1, outport2, outport3. So you can’t really know which output from the FMU is which goto actually. Hope you got the idea from my example. 😊

For any other additional information, please let me know.

Thanks and have a nice day!


Salutări amicale / Best regards,

Calin Tacea

System Design, Integration and PM (RBRO/EPS2)
Robert Bosch SRL | Str. Constanta | No. 30-34 | Zip 400158 | Cluj-Napoca | ROMANIA | www.bosch.com.ro
Calin.Tacea@ro.bosch.com<mailto:Calin.Tacea@ro.bosch.com>

Registered Office: Horia Măcelariu 30-34, sector 1, 013937- Bucureşti, Registration Court: J40/ 7601/ 1994
Managing Directors: Director general : Mihai Boldijar, Director economic: Mafteiu Karla

​

**Comment by TaceaCalin1** [comment]
I am trying to upload PNGs with an example, but it just doesn't work so I will try my best to explain.

In TestSubsystemNoFMU (subsystem) I placed 3 froms, 3 gains and 3 terminators, just as an example. In TestSubsystem(subsystem) I have 3 constants to 3 gotos (named NameOfGotoTag, NameOfGotoTag2, NameOfGotoTag3). This 3 gotos are set to global and have as corresponding blocks the 3 forms from the TestSubsystemNoFMU. The problem is that when I create the FMU and load it in an FMU Block, the names of the outports don’t have the name of the gotos. They are named by default with outport1, outport2, outport3. So you can’t really know which output from the FMU is which goto actually. Hope you got the idea from my example and if I will be allowed to add some screenshots, I will try later.


For any other additional information, please let me know.

Thanks and have a nice day!

**Comment by t-sommer** [comment]
Does the problem still exist with [v3.0-alpha.1](https://github.com/CATIA-Systems/FMIKit-Simulink/releases/tag/v3.0-alpha.1)?

**Comment by TaceaCalin1** [comment]
I just tested. Same problem. The 'Goto' name is: 'TestNameForTags', but when I load the .fmu, the output port is named: 'Outport1'

**Comment by TaceaCalin1** [comment]
Yes. Goto tag name is not taken and when I load the .fmu the name is 'Outport1' instead of 'TestGotoName' (name of the goto tag that I used to test this)
